src/file_pipe_log/pipe.rs

```diff
@@ -155,12 +156,22 @@ impl<F: FileSystem> SinglePipe<F> {
         Ok(())
     }
 
-    fn new_file(&self, seq: FileSeq) -> Result<(PathId, F::Handle)> {
+    /// Recycles one obsolete file from the recycled file list.
+    fn recycle_file(&self) -> (usize, Option<File<F>>) {
+        let mut recycled_files = self.recycled_files.write();
+        let f = recycled_files.pop_front();
+        (recycled_files.len(), f)
+    }
+
+    /// Creates a new log file according to the given [`FileSeq`].
+    ///
+    /// If `recycle_file` is valid, it will reuse it as priority.
+    fn new_file(&self, seq: FileSeq, recycle_file: Option<File<F>>) -> Result<(PathId, F::Handle)> {
         let new_file_id = FileId {
             seq,
             queue: self.queue,
         };
-        if let Some(f) = self.recycled_files.write().pop_front() {
```

Comment:
I just realized there's a bug at L183? If we fail to reuse file, the dst_path should be empty, fs.open will fail.

Reply:
> I just realized there's a bug at L183? If we fail to reuse file, the dst_path should be empty, fs.open will fail.

By the definition of `reuse, `reuse` just use a reference of `dst_path`, it should not be empty. Right ?

Reply:
> I just realized there's a bug at L183? If we fail to reuse file, the dst_path should be empty, fs.open will fail.

Precisely, if we fail to reuse it, the code path will go through `L186` to create a new file according to the `dst_path`. Only if `reuse` success, the code path will go to L183 (open it). So, this case should not exist, right?

Reply:
You are right, I was being silly.